roofit/roofit/CMakeLists.txt

```diff
@@ -148,6 +148,52 @@ ROOT_STANDARD_LIBRARY_PACKAGE(RooFit
     MathCore
 )
 
+if (roofit-arch-libs)
+  ROOT_LINKER_LIBRARY(RooFitCompute_SSE4.1
+      src/RooFitComputeLib.cxx
+    TYPE
+      SHARED
+    DEPENDENCIES
+      RooFit
+  )
+
+
+  ROOT_LINKER_LIBRARY(RooFitCompute_AVX2
+      src/RooFitComputeLib.cxx
+    TYPE
+      SHARED
+    DEPENDENCIES
+      RooFit
+  )
+
+  ROOT_LINKER_LIBRARY(RooFitCompute_AVX512f
+      src/RooFitComputeLib.cxx
+    TYPE
+      SHARED
+    DEPENDENCIES
+      RooFit
+  )
+
+
+  # Add -fno-signaling-nans, -fno-trapping-math and -O3 to enable autovectorization
+  # of the RooFitComputeLibrary (especially needed for GCC).
+  set(common-flags $<$<OR:$<CONFIG:Release>,$<CONFIG:RelWithDebInfo>>:$<$<CXX_COMPILER_ID:GNU>:-fno-signaling-nans> -fno-trapping-math -O3>)
+  target_compile_options(RooFitCompute_SSE4.1 PRIVATE ${common-flags})
+  target_compile_options(RooFitCompute_AVX2 PRIVATE ${common-flags})
+  target_compile_options(RooFitCompute_AVX512f PRIVATE ${common-flags})
+
+  target_compile_options(RooFitCompute_SSE4.1 PRIVATE -msse4 -DRF_ARCH=SSE4)
+  target_compile_options(RooFitCompute_AVX2 PRIVATE -mavx2 -DRF_ARCH=AVX2)
+  target_compile_options(RooFitCompute_AVX512f PRIVATE -mavx512f -DRF_ARCH=AVX512F)
```

Comment:
Can you look up what skylake supports in addition to avx512? We can probably give it a bit more.